### DISCUSSION TITLE
EmotionCssPropInternal Rule of Hooks/Rendering Issue

### DISCUSSION THREAD

**Discussion by thewolff** [emotion-js/emotion#2754] [Q&A]
**Current behavior:**

**To reproduce:**

1. I imported emotion into my Storybook to build some components, so far so good.
2. Bundled my components up as an npm package and received the following error:
```
react.development.js:1433 Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (react.development.js:1433:1)
    at useContext (react.development.js:1441:1)
    at index.es.js:3572:1
    at renderWithHooks (react-dom.development.js:15844:1)
    at updateForwardRef (react-dom.development.js:17532:1)
    at beginWork$1 (react-dom.development.js:19431:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:365:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:414:1)
    at invokeGuardedCallback (react-dom.development.js:468:1)
    at beginWork$$1 (react-dom.development.js:24599:1)
The above error occurred in the <EmotionCssPropInternal> component:
    in EmotionCssPropInternal
    in Box (created by LocationsIndex)
    in div (created by LocationsIndex)
    in LocationsIndex (created by Route)
    in Route (created by Locations)
    in Switch (created by Locations)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by Locations)
    in Locations
```

**Expected behavior:**

My build to work :(

**Environment information:**

- `react` version: 16.0.9
- `@emotion/react` version: 11.9.0

**Comment by srmagura** [comment]
Most likely, you have two instances of `react` in your `node_modules`. Does your library declare a React as a dependency or peer dependency? It should be a peer dependency.

If your library is public on npm, please tell us the name of the package.

**Comment by thewolff** [comment]
It's a private lib unfortunately, though you were spot on about changing React to a peer dependency. Unfortunately, now I'm running into `Module not found: Error: Can't resolve 'react/jsx-runtime'` when pulling our lib in.

**Reply by Andarist** [reply]
Try to create a minimal repro and then share it with us. Without that you are expecting us to read this from tea leaves...
